### PR TITLE
feat: [Draft] added sd+jwt header for SD-JWT

### PR DESCRIPTION
**Pull request by peppelinux**
Definition of these new JWT headers with normative language.
Closes https://github.com/oauthstuff/draft-selective-disclosure-jwt/issues/60

### CODE REVIEW

File: main.md [hunk 1]
```diff
@@ -257,6 +257,11 @@ described in the following, and MAY contain a holder's public key or a reference
 thereto, as well as further claims such as `iss`, `iat`, etc. as defined or
 required by the application using SD-JWTs.
 
+The SD-JWTs MUST be explicitly typed, by setting the `typ` header 
+parameter to `sd+jwt`. This is done to prevent cross-JWT 
+confusion (see "RFC8725", section 3.11).
+
+
```
Comment (Sakurann) [inline main.md:264]:
```suggestion
A `typ` parameter in the JOSE header of an SD-JWT MUST be set to the value `sd+jwt`.
```

Also can you please update a validation section with something like this?

```
 4. Validate the SD-JWT:
    1. Check that the `typ` JOSE header parameter has the value sd+jwt
    2. Ensure that a signing algorithm was used that was deemed secure for the application. Refer to [@RFC8725], Sections 3.1 and 3.2 for details.
    3. Validate the signature over the SD-JWT. 
    4. Validate the issuer of the SD-JWT and that the signing key belongs to this issuer.
    5. Check that the SD-JWT is valid using `nbf`, `iat`, and `exp` claims, if provided in the SD-JWT.
    6. Check that the claim `sd_digests` is present in the SD-JWT.
    7. Check that the `hash_alg` claim is present and its value is understand
       and the hash algorithm deemed secure.
```

File: main.md [hunk 2]
```diff
@@ -503,6 +508,10 @@ When the holder sends the SD-JWT-R to the Verifier, the SD-JWT-R MUST be a JWS
 represented as the JWS Compact Serialization as described in 
 Section 7.1 of [@!RFC7515].
 
+The SD-JWT-R MUST be explicitly typed, by setting the `typ` header 
+parameter to `sd-r+jwt`. This is done to prevent cross-JWT 
+confusion (see "RFC8725", section 3.11).
+
```
Comment (Sakurann) [inline main.md:514]: Note that SD-JWT-R is not always signed.

```suggestion
When SD-JWT-R is signed, a `typ` parameter in the JOSE header of a signed SD-JWT MUST be set to the value `sd-r+jwt`.
```

**Review by Sakurann — CHANGES_REQUESTED** [review]
I would be more comfortable adding only SD+JWT typ for now, since we still need to agree on SD-JWT-R being a JWT vs an object and its validation logic.

**Comment by peppelinux** [comment]
> I would be more comfortable adding only SD+JWT typ for now, since we still need to agree on SD-JWT-R being a JWT vs an object and its validation logic.

Ok.
If sd-jwt-r would be an object and not a jwt how we could bind the sub_jwk of sd-jwt, issued by a trusted party (vci) to the selfsigned release issued by the holder?

**Comment by Sakurann** [comment]
SD-JWT-R is an object when there is no holder binding ie no`sub_jwk` inside SD-JWT.

**Comment by bc-pi** [comment]
see https://github.com/oauthstuff/draft-selective-disclosure-jwt/issues/60#issuecomment-1167745219

**Review by b---c — CHANGES_REQUESTED** [review]
shouldn't do this

**Comment by peppelinux** [comment]
> SD-JWT-R is an object when there is no holder binding ie no`sub_jwk` inside SD-JWT.

mmmhh ... anyway it doesn't have any sense due to the possibility that the SD-JWT can be stolen. The signature and the binding of the sub jwk is a concrete proof of possession

File: main.md (hunk 1 above)
Comment (peppelinux) [reply main.md]: Done here
https://github.com/oauthstuff/draft-selective-disclosure-jwt/pull/83/commits/a4e455d62a6baaa879739975fd50f22256927da6

thank you, it was really important

**Comment by peppelinux** [comment]
SD-JWR-R typ header removed here
https://github.com/oauthstuff/draft-selective-disclosure-jwt/pull/83/commits/b89b4aca8f46cafc7ebfebe69003cf8253617e1f

**Comment by peppelinux** [comment]
Not of interest anymore

**Pull request closed without merging**